components/_patterns/02-molecules/card/01-card.twig

```diff
@@ -1,51 +1,126 @@
-<div class="card{% for modifier in card_modifiers %} card--{{ modifier }}{% endfor %}{% if card_blockname %} {{ card_blockname }}__card{% endif %}">
-  {% if card_img %}
-    <div class="card__img">
-      <img src="{{ card_img }}" />
-    </div>
-  {% elseif image %}
-    <div class="card__img">
-      {{ image }}
-    </div>
+{#
+/**
+ * Available variables:
+ * - card_base_class - base classname of the wrapper
+ * - card_modifiers - array of modifiers to add to the base classname of the wrapper
+ * - card_blockname - blockname prepended to the base classname of the wrapper(s) and each component
+ *
+ * - card_label_base_class - base classname of the card label
+ * - card_label - the card label
+ *
+ * - card_caption_base_class - base classname of the caption wrapper
+ *
+ * - card_img_src - the actual image file location
+ * - card_img_alt - (optional) the alt text for screen readers and when the image cannot load
+ * - card_img_base_class - base classname of the image component. Defaults to "image"
+ * - card_img_modifiers - array of modifiers to add to the base classname of the image
+ *
+ * - card_title - the content of the title component
+ * - card_title_link - (optional) the url the title should link to
+ * - card_title_base_class - base classname of the title component. Defaults to "title"
+ * - card_title_modifiers - array of modifiers to add to the base classname of the title
+ *
+ * - card_subtitle - the content of the subtitle component
+ * - card_subtitle_link - (optional) the url the subtitle should link to
+ * - card_subtitle_base_class - base classname of the subtitle component. Defaults to "subtitle"
+ * - card_subtitle_modifiers - array of modifiers to add to the base classname of the subtitle
+ *
+ * - card_body - the content of the body component
+ * - card_body_base_class - base classname of the body component. Defaults to "body"
+ * - card_body_modifiers - array of modifiers to add to the base classname of the body
+ *
+ * - card_link_text - the content of the link component
+ * - card_link_url - the url the link should link to
+ * - card_link_base_class - base classname of the link component. Defaults to "link"
+ * - card_link_attributes - array of attribute,value pairs for the link attribute
+ * - card_link_modifiers - array of modifiers to add to the base classname of the link
+ *
+ * - card_button_content - the content of the button component
+ * - card_button_url - the url the button should link to
+ * - card_button_base_class - base classname of the button component. Defaults to "button"
+ * - card_button_attributes - array of attribute,value pairs for the button attribute
+ * - card_button_modifiers - array of modifiers to add to the base classname of the button
+ */
+#}
+<div class="{{ card_base_class|default('card') }}{% for modifier in card_modifiers %} {{ card_base_class|default('card') }}--{{ modifier }}{% endfor %}{% if card_blockname %} {{ card_blockname }}__{{ card_base_class|default('card') }}{% endif %}">
```

Comment:
Seems that we are writing `card_base_class|default('card')` over and over again.  I think we can do something like:

`{% set card_base_class_var = card_base_class|default('card') %}`

EDIT: Tested this and it looks like it is working.